### PR TITLE
fix add binary variable

### LINKED ISSUE #244: Knitro.jl cannot handle Integer Variables
```
using JuMP
using KNITRO

model = Model(KNITRO.Optimizer)
@variable(model, x, Bin)
@objective(model, Max, x)
optimize!(model)
print(value(x))
```
Produce this:
```
ERROR: KN_add_con*() must be called before KN_get_var_lobnds().
ERROR: KN_add_con*() must be called before KN_get_var_upbnds().
0.0
```

### PR DESCRIPTION
**Pull request by Thiago-NovaesB**
Close #244

### CODE REVIEW

**Review by odow — COMMENTED** [review]
Can we add the example as a test?

**Comment by thiagonovaesb** [comment]
As this fix was for interacting with JuMP (there were already MINLP tests in the package without using JuMP), I added it to the "example/jump" folder.

**Comment by odow** [comment]
Can you instead/also add it to `MOI_wrapper.jl`?

I didn't test, but something like this should work:
```julia
function test_zero_one_with_no_bounds()
    model = MOI.instantiate(KNITRO.Optimizer)
    x = MOI.add_variable(model)
    MOI.add_constraint(model, x, MOI.ZeroOne())
    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
    MOI.set(model, MOI.ObjectiveFunction{typeof(x)}(), x)
    MOI.optimize!(model)
    @test isapprox(MOI.get(model, MOI.VariablePrimal(), x), 1.0; atol = 1e-6)
    return
end
```

**Comment by Thiago-NovaesB** [comment]
@odow I added the test you asked for, but I'm not sure I did it in the best place.

In addition, I found 2 strange behaviors in the tests:
1) Not all of them pass
image

2) KNITRO.jl doesn't seem to be calling `KN_free `correctly. It must have created some model, did not free the license and tried to create another one. That way you need more than one license to run the local tests.
image

Do you think we would create another issue to discuss these points?

File: test/MOI_wrapper.jl [hunk 1]
```diff
@@ -89,6 +89,18 @@ function test_MOI_Test_cached()
...
 
+function test_zero_one_with_no_bounds()
+    model = MOI.instantiate(KNITRO.Optimizer)
+    MOI.set(model, MOI.Silent(), true)
+    x = MOI.add_variable(model)
+    MOI.add_constraint(model, x, MOI.ZeroOne())
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(model, MOI.ObjectiveFunction{typeof(x)}(), x)
+    MOI.optimize!(model)
+    @test isapprox(MOI.get(model, MOI.VariablePrimal(), x), 1.0; atol = 1e-6)
```
Comment (odow) [inline test/MOI_wrapper.jl:100]:
```suggestion
    @test isapprox(MOI.get(model, MOI.VariablePrimal(), x), 1.0; atol=1e-6)
```

**Comment by odow** [comment]
I changed where the test went. @joaquimg might have an idea about the license. Does it happen every time you run the tests?

**Comment by odow** [comment]
Merging since this is an improvement. I haven't checked CI because this is a fork, so I'll wait to see how `master` runs before tagging, etc.

**Pull request merged**